pkg/api/auth.go

```diff
@@ -253,6 +254,32 @@ func (a *API) authMiddleware(allowedUsers []string) gin.HandlerFunc {
 		}
 
 		c.Set("user", user)
+		c.Set("token", session.AccessToken)
+		c.Next()
+	}
+}
+
+func (a *API) adminAuthMiddleware() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		value, _ := c.Get("user")
+		user, pass := value.(*auth.User)
+		if !pass {
+			a.log.Error("User info missing")
+			c.Redirect(http.StatusSeeOther, "/")
+			return
+		}
+		value, _ = c.Get("token")
+		token, pass := value.(string)
+		userGroups, err := a.azureClient.GroupsForUser(token, user.Email)
+		if err != nil {
+			a.log.WithError(err).Error("problem getting team")
+			c.Redirect(http.StatusSeeOther, "/")
+			return
+		}
+		if !userGroups.Contains("nada@nav.no") {
+			c.Redirect(http.StatusSeeOther, "/")
+			return
+		}
```

Comment:
Could these lines be merged into on function? Returns a bool and either you have access or either you don't. And the new function can do the work of checking for groups?
Then we could have the same action for if your not authenticated or if we have problem connecting to Azure.

Reply:
ok